FAQ.md

```diff
@@ -41,6 +33,82 @@ At the same time, it also plays the role of peer, which can transfer data betwee
 
 ## What is dfdaemon
 
-Dfdaemon is only used for pulling images. It establishes a proxy between dockerd/pouchd and registry.
+Dfdaemon is a local long-running process which is used for translating images pulling request from container engine. It establishes a proxy between container engine and registry.
 
 Dfdaemon filters out layer fetching requests from all requests send by dockerd/pouchd when pulling images, then it uses dfget to downloading these layers.
+
+## What is the sequence of supernode's CDN functionality and P2P distribution
+
+When dfget starts to pull an image which has not been cached in supernode yet, supernode will do as the following sequence:
+
+- First Step: supernode triggers the downloading task:
+    - fetch the file/image length;
+    - divide the length into pieces;
+    - start to download the file piece by piece;
+- Second Step: 
+    - supernode finishes to download one piece
+    - supernode starts the P2P distribution work for the downloaded one piece among peers;
+    
+In a word, supernode does not have to wait for all the piece downloadings finished, it can concurrently start piece downloading once one piece downloaded.
+
+## What is the peer scheduling algorithm of supernpde when many peers start to pull the same file
+
+
+## What is the size of block(piece) when distribution
+
+Dragonfly tries to make block(piece) size dynamically to ensure efficiency.
+
+The size of pieces which is calculated as per the following strategy:
+
+- If file's total size is less than 200MB, then the piece size is `4MB` by default.
+- Otherwise, it equals to `min{ totalSize/100MB + 2 MB, 15MB }`.
+
+## What is the difference between Dragonfly's P2P algorithm and bit-torrent
```

Comment:
Could you help to answer this question? @garfield009 @lowzj 
I think many peopler are curious about the answer.